packages/itwinui-react/src/core/ColorPicker/ColorBuilder.tsx

```diff
@@ -253,37 +265,49 @@ export const ColorBuilder = React.forwardRef((props, ref) => {
       {...rest}
     >
       <Box
-        className='iui-color-field'
+        as='div'
+        {...colorFieldProps}
+        className={cx('iui-color-field', colorFieldProps?.className)}
         style={
           {
             '--iui-color-field-hue': hueColorString,
             '--iui-color-picker-selected-color': dotColorString,
+            ...colorFieldProps?.style,
           } as React.CSSProperties
         }
-        ref={squareRef}
-        onPointerDown={(event: React.PointerEvent) => {
-          event.preventDefault();
-          updateSquareValue(event, 'onClick');
-          setColorDotActive(true);
-          colorDotRef.current?.focus();
-        }}
+        ref={useMergedRefs(squareRef, colorFieldProps?.ref)}
+        onPointerDown={mergeEventHandlers(
+          (event: React.PointerEvent) => {
+            event.preventDefault();
+            updateSquareValue(event, 'onClick');
+            setColorDotActive(true);
+            colorDotRef.current?.focus();
+          },
+          colorFieldProps?.onPointerDown,
+        )}
       >
         <Box
-          className='iui-color-dot'
+          as='div'
+          {...colorDotProps}
+          className={cx('iui-color-dot', colorDotProps?.className)}
           style={
             {
               '--iui-color-dot-inset-block': `${squareTop.toString()}% auto`,
               '--iui-color-dot-inset-inline': `${squareLeft.toString()}% auto`,
+              ...colorDotProps?.style,
             } as React.CSSProperties
           }
-          onPointerDown={() => {
-            setColorDotActive(true);
-            colorDotRef.current?.focus();
-          }}
+          onPointerDown={mergeEventHandlers(
+            () => {
+              setColorDotActive(true);
+              colorDotRef.current?.focus();
+            },
+            colorDotProps?.onPointerDown,
+          )}
           onKeyDown={handleColorDotKeyDown}
           onKeyUp={handleColorDotKeyUp}
```

Comment:
why are `onKeyDown` and `onKeyUp` handlers not merged with the corresponding props?

Reply:
I must have missed these. I have used `mergeEventHandlers` for these handlers.